File: src/main/java/me/mrlion/lionplugin/commands/MsgCommand.java
package me.mrlion.lionplugin.commands;

import me.mrlion.lionplugin.LionPlugin;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Objects;

public class MsgCommand implements CommandExecutor {
    private final LionPlugin plugin;

    public MsgCommand(LionPlugin plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!(sender instanceof Player)) {
            sender.sendMessage(ChatColor.RED + "This command can't be used by console!");
            return true;
        }

        if (args.length < 2) {
            String msgUsage = Objects.requireNonNull(plugin.getConfig().getString("msg.usage"));
            sender.sendMessage(ChatColor.translateAlternateColorCodes('&', msgUsage));
            return true;
        }

        Player recipient = Bukkit.getPlayer(args[0]);
        if (recipient == null) {
            String playerNotFound = Objects.requireNonNull(plugin.getConfig().getString("msg.player-not-found"));
            sender.sendMessage(ChatColor.translateAlternateColorCodes('&', playerNotFound));
            return true;
        }

        StringBuilder messageBuilder = new StringBuilder();
        for (int i = 1; i < args.length; i++) {
            messageBuilder.append(args[i]).append(" ");
        }
        String message = messageBuilder.toString().trim();

        String outgoingMessage = Objects.requireNonNull(plugin.getConfig().getString("msg.outgoing-message-format"))
                .replace("{sender}", sender.getName())
                .replace("{receiver}", recipient.getName())
                .replace("{message}", message);
        sender.sendMessage(ChatColor.translateAlternateColorCodes('&', outgoingMessage));

        String incomingMessage = Objects.requireNonNull(plugin.getConfig().getString("msg.incoming-message-format"))
                .replace("{sender}", sender.getName())
                .replace("{receiver}", recipient.getName())
                .replace("{message}", message);
        recipient.sendMessage(ChatColor.translateAlternateColorCodes('&', incomingMessage));

        return true;
    }
}

File: src/main/java/me/mrlion/lionplugin/commands/MsgCommand.kt
package me.mrlion.lionplugin.commands

import me.mrlion.lionplugin.LionPlugin
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*

class MsgCommand(private val plugin: LionPlugin) : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (sender !is Player) {
            sender.sendMessage(ChatColor.RED.toString() + "This command cant be used by console!")
            return true
        }

        val player = sender

        if (args.size < 2) {
            val msgUsage = Objects.requireNonNull(plugin.config.getString("msg.usage"))?.let { ChatColor.translateAlternateColorCodes('&', it) }
            player.sendMessage(msgUsage)
            return true
        }

        val recipient = Bukkit.getPlayer(args[0])
        if (recipient == null) {
            val playerNotFound = Objects.requireNonNull(plugin.config.getString("msg.player-not-found"))?.let { ChatColor.translateAlternateColorCodes('&', it) }
            player.sendMessage(playerNotFound)
            return true
        }

        val message = StringBuilder()
        for (i in 1 until args.size) {
            message.append(args[i]).append(" ")
        }

        val outgoingMessage = Objects.requireNonNull(plugin.config.getString("msg.outgoing-message-format"))
                ?.replace("{sender}", player.name)
                ?.replace("{receiver}", recipient.name)?.let {
                    ChatColor.translateAlternateColorCodes('&', it
                            .replace("{message}", message.toString()))
                }
        player.sendMessage(outgoingMessage)

        val incomingMessage = Objects.requireNonNull(plugin.config.getString("msg.incoming-message-format"))
                ?.replace("{sender}", player.name)
                ?.replace("{receiver}", recipient.name)?.let {
                    ChatColor.translateAlternateColorCodes('&', it
                            .replace("{message}", message.toString()))
                }
        recipient.sendMessage(incomingMessage)

        return true
    }
}